Integration_App3/Part3_App3/src/WeatherDataSubject.java

```diff
@@ -0,0 +1,77 @@
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class WeatherDataSubject {
+    
+    private List<WeatherDataObserver> observers ;
+
+// private int state;
+
+//Celsius Stored in the data base
+    private double temperature;
+
+    private double humidity;
+    private double windSpeed;
+    private String precipitationType;
+
+    public WeatherDataSubject(double temperature, double humidity, double windSpeed, String precipitationType) {

```

Comment:
this code breaks SRP, instead, you had to break this subject to multiple subjects